server/src/test/java/e2e/ConditionalsTest.java

```diff
@@ -318,6 +321,57 @@ private static Stream<Arguments> provideSuccessArguments() {
         }
     }
 
+    // TODO: After ListTaskRunRequest is available, add NestedIf tests.
+
+    @Nested
+    class NestedIf {
+        @ParameterizedTest
+        @MethodSource("provideSuccessArguments")
+        void shouldCompleteNotInWorkflowWithConditionals(InputObj inputObject, boolean expectedOutput) {
+            workflowVerifier
+                    .prepareRun(workflowNotIn, Arg.of("input", inputObject))
+                    .waitForStatus(LHStatus.COMPLETED)
+                    .thenVerifyTaskRunResult(0, 1, variableValue -> assertThat(variableValue.getBool())
+                            .isEqualTo(true))
+                    .thenVerifyTaskRunResult(0, 3, variableValue -> assertThat(variableValue.getBool())
+                            .isEqualTo(expectedOutput))
+                    .start();
+        }
+
+        private static Stream<Arguments> provideSuccessArguments() {
```

Comment:
This wont work. verify test method arguments, it is expecting an instance of `InputObj` class